app-sso/service-operators/tls-scenario-guides.hbs.md

```diff
@@ -0,0 +1,785 @@
+kind: AuthServer
+metadata:
+  name: sso
+  namespace: login
+  annotations:
+    sso.apps.tanzu.vmware.com/allow-unsafe-identity-provider: ""
+    sso.apps.tanzu.vmware.com/allow-client-namespaces: "*"
+spec:
+  #! --- TLS ---
+  tls:
+    issuerRef:
+      name: letsencrypt-production
+      kind: ClusterIssuer
+  #! -----------
+  identityProviders:
+    - name: test-users
+      internalUnsafe:
+        users:
+          - username: user
+            password: password
+  tokenSignature:
+    signAndVerifyKeyRef:
+      name: signing-key
+
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: RSAKey
+metadata:
+  name: signing-key
+  namespace: login
+spec:
+  secretTemplate:
+    type: Opaque
+    stringData:
+      key.pem: $(privateKey)
+      pub.pem: $(publicKey)
+```
+
+## I have an `Issuer`
+
+This scenario is almost exactly the same as with a `ClusterIssuer`, but `Issuer` is Namespace-scoped and needs to be
+collocated with the `AuthServer`.
+
+```yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: login
+
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: letsencrypt-production
+  namespace: login
+spec:
+  acme:
+    privateKeySecretRef:
+      name: letsencrypt-production
+    server: https://acme-v02.api.letsencrypt.org/directory
+    solvers:
+      - http01:
+          ingress:
+            class: contour
+
+---
+apiVersion: sso.apps.tanzu.vmware.com/v1alpha1
+kind: AuthServer
+metadata:
+  name: sso
+  namespace: login
+  annotations:
+    sso.apps.tanzu.vmware.com/allow-unsafe-identity-provider: ""
+    sso.apps.tanzu.vmware.com/allow-client-namespaces: "*"
+spec:
+  #! --- TLS ---
+  tls:
+    issuerRef:
+      name: letsencrypt-production
+  #! -----------
+  identityProviders:
+    - name: test-users
+      internalUnsafe:
+        users:
+          - username: user
+            password: password
+  tokenSignature:
+    signAndVerifyKeyRef:
+      name: signing-key
+
+---
+apiVersion: secretgen.k14s.io/v1alpha1
+kind: RSAKey
+metadata:
+  name: signing-key
+  namespace: login
+spec:
+  secretTemplate:
+    type: Opaque
+    stringData:
+      key.pem: $(privateKey)
+      pub.pem: $(publicKey)
+```
+
+## I have an existing `Certificate`
+
+A `Certificate` is a Namespace-scoped API provided by [cert-manager](https://cert-manager.io) which represents a TLS
+certificate obtained from a `(Cluster)Issuer`. To create `Certificate`, you need to know the name and kind of your
+issuer.
+
+These scenarios use [Let's Encrypt](https://letsencrypt.org/)'s production API and assume that a
+`ClusterIssuer` by the name `letsencrypt-production` exists.
+See [the `ClusterIssuer` scenario](#i-have-a-clusterissuer-_recommended_) for how to set the issuer up.
+
+When using `Certificate`, its `.spec.dnsNames` must contain the FQDN of the templated issuer URI. Make sure that your
+AppSSO package installation's `domain_name` and `domain_template` are compatible with your DNS name.
```

Comment:
**[suggestion]** Is it worth adding a "troubleshooting" section about what happens when the DNS dnsNames are incorrect?

Reply:
I like that idea. Maybe we can merge this for now and see if we have time before the end of the week to add a troubleshooting section?

Reply:
Yes, good idea!